Annotato/Annotato/ViewModels/Document/PdfViewModel.swift

```diff
@@ -1,14 +1,25 @@
-import UIKit
 import PDFKit
+import AnnotatoSharedLibrary
 
 class PdfViewModel {
     let autoScales = true
     let document: PDFDocument
 
-    init(baseFileUrl: URL) {
-        guard let document = PDFDocument(url: baseFileUrl) else {
+    init(document: Document) {
+        var pdfDocument = PDFDocument(url: document.localFileUrl)
+
+        if pdfDocument == nil,
+           let remoteFileUrlString = document.baseFileUrl,
+           let remoteFileUrl = URL(string: remoteFileUrlString) {
+            pdfDocument = PDFDocument(url: remoteFileUrl)
+        }
+
+        guard let pdfDocument = pdfDocument else {
+            AnnotatoLogger.error("Could not load document")
+            // TODO: Handle failed init more gracefully
             fatalError("No such document")
```

Comment:
hahaha, I guess just log an error then return?

Reply:
> Or is it because the models must have an optional base file url (to cater for offline case), so the fluent entities have to be optional as well.

Yeah more of this.